requests_cache/core.py

```diff
@@ -4,250 +4,182 @@
     """
 
     def __init__(
         self,
         cache_name: str = 'cache',
         backend: str = None,
-        expire_after: Union[int, float, timedelta] = None,
+        expire_after: ExpirationTime = None,
         allowable_codes: Iterable[int] = (200,),
         allowable_methods: Iterable['str'] = ('GET', 'HEAD'),
         filter_fn: Callable = None,
         old_data_on_error: bool = False,
-        **kwargs
+        **kwargs,
     ):
         self.cache = backends.create_backend(backend, cache_name, kwargs)
-        self._cache_name = cache_name
-
-        if expire_after is not None and not isinstance(expire_after, timedelta):
-            expire_after = timedelta(seconds=expire_after)
-        self._cache_expire_after = expire_after
-        self._request_expire_after = 'default'
+        self.allowable_codes = allowable_codes
+        self.allowable_methods = allowable_methods
+        self.filter_fn = filter_fn or (lambda r: True)
+        self.old_data_on_error = old_data_on_error
 
-        self._cache_allowable_codes = allowable_codes
-        self._cache_allowable_methods = allowable_methods
-        self._filter_fn = filter_fn or (lambda r: True)
-        self._return_old_data_on_error = old_data_on_error
-        self._is_cache_disabled = False
+        self._cache_name = cache_name
+        self._expire_after = expire_after
+        self._request_expire_after = None
+        self._disabled = False
 
         # Remove any requests-cache-specific kwargs before passing along to superclass
-        session_kwargs = {k: v for k, v in kwargs.items() if k not in BACKEND_KWARGS}
+        session_kwargs = {k: v for k, v in kwargs.items() if k not in backends.BACKEND_KWARGS}
         super().__init__(**session_kwargs)
 
-    def _determine_expiration_datetime(self, relative_to=None):
-        """Determines the absolute expiration datetime for a response.
-        Requires :attr:`self._cache_expire_after` and :attr:`self._request_expire_after` to be set.
-        See :meth:`request` for more information.
-
-        :param response: the response (potentially loaded from the cache)
-        :type response: requests.Response
-        :param relative_to: Parameter for easy unit testing to fix ``now``,
-                            defaults to ``datetime.now(timezone.utc)`` for normal use.
-        :type relative_to: Union[None, datetime.datetime]
-        :return: The absolute expiration date
-        :rtype: datetime.datetime
+    @property
+    def expire_after(self):
+        """Get either the per-session expiration, or per-request expiration, if set"""
+        return self.request_expire_after or self._expire_after
+
+    @expire_after.setter
+    def expire_after(self, value: ExpirationTime):
+        """Get either the per-session expiration, or per-request expiration, if set"""
+        self._expire_after = value
+
+    @property
+    def request_expire_after(self) -> Optional[ExpirationTime]:
```

Comment:
ExpirationTime is already a `Union[..., None]` (https://github.com/reclosedev/requests-cache/pull/189/files#diff-5ed71c002223078604be949671777da35b8e50b43edf121ad9d8799af397f281R9), so `Optional` is redundant. 
However, using it might make it slightly more clear to a reader that the return value might be None.

Reply:
Yep, you're right, it's redundant here.